dubbo-dependencies-bom/pom.xml

```diff
@@ -316,7 +316,7 @@
         <scope>test</scope>
       </dependency>
       <dependency>
-        <groupId>org.jboss.netty</groupId>
```

Comment:
`io.netty:netty` will conflict with 4.x

Reply:
Fortunately, dubbo-remoting-netty does not rely on io.netty:netty, and it just depends on netty-codec-http2, netty-handler-proxy, netty-transport-classes-epoll, netty-transport-native-epoll which all belongs to netty4.

Reply:
and dubbo-remoting-netty just depends on netty, i tested it at my repository, it seemed that changing group id of netty  from org.jboss.netty to io.netty doesn't conflict with current dubbo codes.
https://github.com/zrlw/dubbo/actions/runs/13623047801

Reply:
> `io.netty:netty` will conflict with 4.x

since netty4.0.0.Final, netty4 used io.netty:netty-all instead of io.netty:netty

Reply:
But changing `dubbo-dependencies-bom` here might be a breaking change.

Reply:
i see. i think the reason that moving netty to netty-all since netty 4.0.0.Final is to allow using them together in one application, there are no conflicts between io.netty:netty and io.netty:netty-all, 
https://mvnrepository.com/artifact/io.netty/netty
![netty](https://github.com/user-attachments/assets/c209e788-055b-4d00-8952-0031717cc53d)
https://mvnrepository.com/artifact/io.netty/netty-all
![netty-all](https://github.com/user-attachments/assets/d55fb197-1a0f-4e97-8f5b-33d10b9a24d2)

Reply:
> But changing `dubbo-dependencies-bom` here might be a breaking change.

yes, it's a breaking change. Comparing to 3.2.10.Final, 3.10.6.Final (2016) had made many changes. defects and risks should be balanced.

Reply:
![Netty 3.2.10.Final](https://github.com/user-attachments/assets/c5672f4a-0689-40c2-9e2d-3c996b68e1b8)

Reply:
You‘re right, but I would prefer that to catch such exception and retry automatically. In the future, maybe 3.4, we can remove all the support of netty3 anymore. It is important to prevent importing surprising changes in minor release.

Reply:
ok